src/Raven.Client/Documents/Operations/ETL/Queue/AzureQueueStorageConnectionSettings.cs

```diff
@@ -0,0 +1,163 @@
+using System;
+using System.Linq;
+using Sparrow.Json.Parsing;
+
+namespace Raven.Client.Documents.Operations.ETL.Queue;
+
+public sealed class AzureQueueStorageConnectionSettings
+{
+    public EntraId EntraId { get; set; }
+
+    public string ConnectionString { get; set; }
+
+    public Passwordless Passwordless { get; set; }
+
+    public bool IsValidConnection()
+    {
+        if (IsOnlyOneConnectionProvided() == false)
+        {
+            return false;
+        }
+
+        if (EntraId != null && EntraId.IsValid() == false)
+        {
+            return false;
+        }
+
+        if (Passwordless != null && Passwordless.IsValid() == false)
+        {
+            return false;
+        }
+
+        return true;
+    }
+    
+    private bool IsOnlyOneConnectionProvided()
+    {
+        int count = 0;
+
+        if (EntraId != null) 
+            count++;
+    
+        if (!string.IsNullOrWhiteSpace(ConnectionString))
+            count++;
+
+        if (Passwordless != null) 
+            count++;
+
+        return count == 1;
+    }
+
+    public string GetStorageUrl()
+    {
+        if (ConnectionString != null)
+        {
+            return GetUrlFromConnectionString(ConnectionString);
+        }
+
+        string storageAccountName = GetStorageAccountName();
+        return $"https://{storageAccountName}.queue.core.windows.net/";
+    }
+    
+    private string GetUrlFromConnectionString(string connectionString)
+    {
+        var parts = connectionString.Split(';')
+            .Select(p => p.Split('='))
+            .ToDictionary(p => p[0], p => p.Length > 1 ? p[1] : "");
```

Comment:
I think you could use `SqlConnectionStringParser.GetConnectionStringValue()` here

Reply:
done